x-pack/plugins/reporting/server/core.ts

```diff
@@ -134,7 +134,7 @@ export class ReportingCore {
     this.getContract = () => ({
       usesUiCapabilities: () => config.roles.enabled === false,
       registerExportTypes: (id) => id,
-      getScreenshots: this.getScreenshots.bind(this),
+      getScreenshots: config.statefulSettings.enabled ? this.getScreenshots.bind(this) : undefined,
```

Comment:
WDYT about raising an error here?

```suggestion
      getScreenshots: config.statefulSettings.enabled ? this.getScreenshots.bind(this) : () => {
        throw new Error('...');
      },
```

Reply:
Im not sure it will be good to throw an error here since it would impact reporting on start for serverless. Maybe log something? 🤔